example-streamer-implementations/README.md

```diff
@@ -0,0 +1,91 @@
+
+## Supported Setups
+
+Here are the setups that can be built with these streamers and different entities with the Zenoh, SomeIP and MQTT transports.
+Depending on which setup you want to try, run the streamer with:
+
+```bash
+cargo run --bin <zenoh_mqtt/zenoh_someip> --features=<check cargo.toml or logs to see which features you need> -- --config="./example-streamer-implementations/README.md"
+```
+
+### Client-Service Setups
+
+In a setup with one client and one service, the service runs in the background while the client periodically makes requests to it.
+Once the server receives a request it will respond with a reply.
+The request message contains information on a "sink" (the URI of the service entity which it tries to reach) and a source (the URI of the client so that the service knows where to send the response to).
+
+For a single setup you can choose either:
+- a Zenoh Client and an MQTT Service (A cars' software requesting information from the cloud)
+- or an MQTT Client and a Zenoh Service (A backend service trying to pull telemetry data from a running car)
+- a Zenoh Client and a SomeIP Service (The infotainment system requesting some mechatronics sensor data)
+- or a SomeIP Client and a Zenoh Service (A mechatronics component asking the infotainment system for input)
+
+### Publish-Subscribe Setups
+
+This setup is more straight forward and consists of one publisher broadcasting messages to a topic and a subscriber who listens to the topic.
+Messages that the publisher sends contain a "topic" that the message will be available on. There is no response expected so publish messages do not contain the publishers URI.
+The subscriber listens to one or multiple topics via a filter.
+
+For this setup you can also choose:
+- a Zenoh Publisher and an MQTT Subscriber (A backend service getting live data from a car)
+- an MQTT Publisher and a Zenoh Subscriber (A car getting over the air traffic information)
+- a Zenoh Publisher and a SomeIP Subscriber (An autoedge app pushing some configurations to a mechatronics component)
+- or a SomeIP Publisher and a Zenoh Subscriber (An infotainment app getting live data from a sensor)
+
+## Understanding the Configuration Files
+
+Reference the `DEFAULT_CONFIG.json5` configuration file to understand the basic configuration options for the Streamer.
+
+The `ZENOH_CONFIG.json5` file is used to set Zenoh configurations. By default, it is only used to set listening endpoints, but can be used with more configurations according to [Zenoh's page on it](https://zenoh.io/docs/manual/configuration/#configuration-files).
+
+The 'static_subscriptions.json' is only needed when you set up a publish-subscribe system and can be ignored for a client-service system.
+
+The 'vsomeip-config/point_to_point.json'
+
+The MQTT Transport has its configuration hardcoded for now. For the settings check main.rs for "mqtt_config".
```

Comment:
Are there pieces that make sense to expose here now? I'm imagining if/when folks try to use this reference streamer for the Eclipse SDV Hackathon it could make it easier.